### DISCUSSION TITLE
Subpoint sun on earth

### DISCUSSION THREAD

**Discussion by powersoft** [skyfielders/python-skyfield#937] [Q&A] [answered]
I have read the documentation, in front of that I made a definition in python/skyfield to calculate the sublat position of the sun on earth. My question is the definition correct, have no clou how to check the results.
Thanks for any help.
Cheers Jan

`def subpoint_sun(time):
    earth, sun = eph['earth'], eph['sun']
    
    astr = earth.at(time).observe(sun)
    ra, dec, distance = astr.apparent().radec()
    earth = 399  # NAIF code for the Earth center of mass
    ra_hours = ra.hours
    dec_degrees = dec.degrees
    zon = position_of_radec(ra_hours, dec_degrees, t=time, center=earth)
    subpoint = wgs84.subpoint(zon)
    return subpoint.latitude.degrees, subpoint.longitude.degrees

**Comment by brandon-rhodes** [comment, marked as answer]
Happily, a quick Google search brings back a number of pages that identify the subsolar point. For example:

https://www.timeanddate.com/worldclock/sunearth.php

Try checking your result against that.

Also, I think you can stop with the `apparent()` call and use that raw position to ask for a latitude and longitude:

https://rhodesmill.org/skyfield/earth-satellites.html#satellite-longitude-latitude-and-height

That would let you skip the step of generating an RA and dec. Good luck, and let us know what you find out!

**Reply by powersoft** [reply]
I have try this code, the first part is working, but the converting to the subpoint will not work. Please can you help me out with this.
`
from skyfield.api import load, wgs84
ts = load.timescale()
eph = load('de421.bsp')
sun, moon, earth = eph['sun'], eph['moon'], eph['earth']
utc_time = ts.now()

geocentric = moon.at(utc_time)
print(geocentric.position.au) 
subpoint=wgs84.subpoint(geocentric)
`

**Reply by brandon-rhodes** [reply]
If you will try doing a `print()` of your `moon.at(utc_time)` result, you will see that it is not a geocentric position. Try what you did earlier: ask the earth `at()` a given time to `observe()` the Moon.

**Reply by powersoft** [reply]
Thanks for this, it is now simple. And the code is nice streamed

**Reply by brandon-rhodes** [reply]
Thanks for checking back in, I'm glad it works now!
